subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -46,7 +46,13 @@ TBD
 
 === Potential breaking changes
 
-TBD
+==== User provided daemon JVM arguments are merged with defaults
+
+Prior to 8.0, setting `org.gradle.jvmargs` would overwrite all default daemon JVM args, however now Gradle's default JVM arguments are merged with the arguments provided by this property. In some cases, this may cause builds which set `org.gradle.jvmargs` to have different memory configuration in Gradle 7.6.
+
+As it was before, the default daemon JVM args are `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`. Previously, if `org.gradle.jvmargs` is set to `-DsomeProperty=123`, the existing behavior would be to resolve the daemon arguments to `-DsomeProperty=123` while the new behavior would resolve to `-DsomeProperty=123 -Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`.
+
+Since the JVM's default memory arguments are more generous than Gradle's default daemon JVM arguments, this change can cause existing builds to fail with an `OutOfMemoryError` if the build uses more memory than is explicitly set in `org.gradle.jvmargs`.
```

Comment:
```suggestion
==== User provided daemon JVM arguments merge with defaults

Prior to 8.0, setting `org.gradle.jvmargs` overwrites all default daemon JVM args. In 8.0 and above, Gradle merges default JVM arguments with the arguments provided by this property. This can effect the memory configuration of builds that set `org.gradle.jvmargs`.

Consider the default daemon JVM arguments:

----
-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError
----

Prior to 8.0, if a user assigned `org.gradle.jvmargs` the value `-DsomeProperty=123`, Gradle resolved the daemon arguments to the following:

----
-DsomeProperty=123
----

In 8.0 and above, if a user assigns `org.gradle.jvmargs` the value `-DsomeProperty=123`, Gradle resolves the daemon arguments to the following:

----
-DsomeProperty=123 -Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`
----

Since the JVM's default memory arguments are more generous than Gradle's default daemon JVM arguments, this change can cause existing builds to fail with an `OutOfMemoryError` if the build uses more memory than explicitly set in `org.gradle.jvmargs`.
```

Suggestion: use the terms "Before 8.0" and "In 8.0 and above" to make the subject of sentences (and comparisons) clearer.

Separate large collections of JVM arguments into block code snippets to make them more readable.

Use active voice when possible to make it clearer when Gradle behaves in some way, and when the user behaves in some way. I find the docs a little hard to parse as written.